free-1.md

```diff
@@ -10,41 +10,120 @@ _What need is met by your idea? Who are you helping? What is that person's
+
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
-like? Why is this the right way to interact with the problem domain?_ 
+like? Why is this the right way to interact with the problem domain?_
+
+I want to keep to commands as simple as possible, so there would be a small
+set of defined shapes, which the user would instantiate as objects. The 
+properties of the object could be modified (i.e. color and size), and the
+objects could be easily modified. There would also be a set of animation
+methods, which could allow shapes to float across the screen, rotate in place,
+and gradually change size/color. The user would call methods on the objects 
+and they could specifiy the intervals at which the methods would be called.
+I imagine the interface looking similar to ContextFree, where the user writes 
+code in one part of the window, and the result of compiled code appears next 
+to it. This would be a good way for the user to immediately see the results
+of their animations, and modify them as necessary.
 
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+The user could incrementally see the progress of their animations, by compiling
+the code and the result being displayed right next to it. This gives the user
+the immediate satisfaction of being able to see their coded creations. It is 
+likely that the user will make syntax errors, perhaps having invalid parameters 
+for the shapes and methods. Errors will be identified at compile time; the 
+animations won't be generated if there are syntax errors, which would hopefully 
+be highlighted in the code. 
 
 ### Expressiveness
 _What should be easy to do in this language? What should be possible, but
 difficult? What should be impossible or very difficult?_
 
+It should be easy to create simply shapes and animate them! There should be
+room to also generate more complex combinations of shapes and animations so
+that more experienced users won't be bored. I imagine it would be possible
+to create less loop-like animations and longer story-like animations, but it
+would take a lot of work and at that point it would be better for the user to
+use established animation software. It would be difficult to do anything other
+than animated art, like solving mathematical functions. It would be impossible
+to do something like parsing text or composing a song.
+
 
 ### Related work
 _Are there any other DSLs in this domain? If not, describe how you know there
 aren't and conjecture why not. If so, describe them and provide links. How well 
 do they address the need? Are there any particularly admirable qualities of the
 language? Are there parts of the language you think could be improved?_
 
+[Processing](https://processing.org/) is a DSL that is very similar to the type 
```

Comment:
I personally don't know anything about this domain, but both of these programs look like great sources of inspiration. Would you want to incorporate drag and drop like Kaleidoscope? Seems like it would require much more effort on your part to code that interface. I agree with the "simplicity of the GUI of Kaleidoscope and the customizable nature of Processing". Maybe what you can do is to incorporate the drag/drop aspect of Processing, but as pieces of code add onto each other, maybe you could have the programming code adding to itself on the side, code that the user can customize? This would also serve as a good intro to programming.